src/snippet.ts

```diff
@@ -47,27 +47,31 @@ export class Snippet {
             this.currNum = num
         }
 
-        let response = await this._doRequest(this.currLanguage)
+        let response = await this._doRequest()
         return { language, data: response.data }
     }
 
-    async loadNext(): Promise<Response> {
+    getNextAnswerNumber(): number {
```

Comment:
That's a better abstraction than what we had before because it adheres to the single-responsibility principle. Good work.